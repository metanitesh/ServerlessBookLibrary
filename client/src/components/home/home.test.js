import React from 'react';
import Home from './home';
import { BrowserRouter } from "react-router-dom";
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'
configure({ adapter: new Adapter() });


it('renders without crashing', () => {
  const item = renderer
     .create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
      ).toJSON();
   expect(item).toMatchSnapshot();
});


it('should redirect if user is authenticated', () => {
  const fnClick = jest.fn();
  const mock = {
    push: fnClick
  }

  renderer
     .create(
      <BrowserRouter>
        <Home authStatus={true} history={mock}/>
      </BrowserRouter>
      ).toJSON();
      expect(fnClick).toHaveBeenCalled();
});

xit('should login when clicked the login button', () => {
                            
  const fnClick = jest.fn();
  const mock = {
    push: fnClick
  }

  const fnClick2 = jest.fn();
  const mock2 = {
    push: fnClick
  }

  const home = mount(
   <BrowserRouter>
     <Home authStatus={true} props={mock} history={mock2}/>
   </BrowserRouter>
   )

   home
   .find('#login')
   .simulate('click');

   
   expect(fnClick).toHaveBeenCalled();
});




