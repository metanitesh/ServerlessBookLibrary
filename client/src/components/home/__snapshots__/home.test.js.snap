// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <header
    className="header"
  >
    <a
      href="/library"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo"
        src="/Blinkist-Logo.png"
      />
    </a>
    <h6>
       User 
      Log in
    </h6>
  </header>,
  <main
    className="main"
  >
    <form
      onSubmit={[Function]}
    >
      <button
        className="button"
        id="login"
      >
        Login
      </button>
    </form>
  </main>,
]
`;
